pkg/policy/k8s/service_test.go

```diff
@@ -443,14 +446,41 @@ func TestPolicyWatcher_updateToServicesPolicies(t *testing.T) {
 		},
 	})
 
-	// Add baz-svc, which is not selectable and thus must not trigger a policyAdd
+	// Add baz-svc, which is selected by svcByLabelCNP
 	svcCache[bazSvcID] = fakeService{
 		svc: bazSvc,
 		eps: bazEps,
 	}
 	err = p.updateToServicesPolicies(bazSvcID, bazSvc, nil)
 	assert.NoError(t, err)
-	assert.Empty(t, policyAdd)
+	rules = <-policyAdd
```

Comment:
Can you validate the following cases:

1. Add policy with non-matching service selector
2. Add service that matches, validate selector is correct
3. Update matched service, validate selector is correct
4. Add second matching service, validate selector
5. Delete each service, validating generated selector
6. Delete policy
7. Add service
8. Add policy, validate selection is correct

Make sense?

Reply:
Done.